cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/spec/StageSpec.java

```diff
@@ -140,15 +147,16 @@ public Builder(String name, PluginSpec plugin) {
       this.plugin = plugin;
       this.inputs = new HashSet<>();
       this.outputs = new HashSet<>();
+      this.inputSchemas = new HashMap<>();
     }
 
     public Builder setErrorDatasetName(String errorDatasetName) {
       this.errorDatasetName = errorDatasetName;
       return this;
     }
 
-    public Builder setInputSchema(Schema inputSchema) {
-      this.inputSchema = inputSchema;
+    public Builder addInputSchemas(Map<String, Schema> inputSchemas) {
```

Comment:
would be nice to add a `addInputSchema(String, Schema)` method for convenience, so we don't have to always create a map in the unit tests.